lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -297,9 +302,50 @@ def grouper(iterable, chunk_size=self.USER_BATCH_SIZE, fillvalue=None):
             args = [iter(iterable)] * chunk_size
             return izip_longest(*args, fillvalue=fillvalue)
 
-        users = CourseEnrollment.objects.users_enrolled_in(context.course_id, include_inactive=True)
-        users = users.select_related('profile')
-        return grouper(users)
+        def users_for_course(course_id):
+            """
+            Get all the enrolled users in a course.
+
+            This method fetches & loads the enrolled user objects at once which may cause
+            out-of-memory errors in large courses. This method will be removed when
+            `OPTIMIZE_GET_LEARNERS_FOR_COURSE` waffle flag is removed.
```

Comment:
No code change required now, I would just make sure there's a ticket in the backlog to rip out this old function once it's confirmed that the new version works (and you could rename the new version at that time).

Reply:
Thank you @iloveagent57 I will create a new ticket once the fix is tested on production.